processing/src/main/java/io/druid/segment/DimensionMerger.java

```diff
@@ -0,0 +1,165 @@
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import com.google.common.io.OutputSupplier;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.Buffer;
+import java.nio.IntBuffer;
+import java.util.List;
+
+/**
+ * Processing related interface
+ *
+ * A DimensionMerger is a per-dimension stateful object that encapsulates type-specific operations and data structures
+ * used during the segment merging process (i.e., work done by IndexMerger/IndexMergerV9).
+ *
+ * This object is responsible for:
+ *   - merging any relevant structures from the segments (e.g., encoding dictionaries)
+ *   - writing the merged column data and any merged indexing structures (e.g., dictionaries, bitmaps) to disk
+ *
+ * @param <EncodedType> class of the encoded values
+ * @param <ActualType> class of the actual values
+ */
+public interface DimensionMerger<EncodedType extends Comparable<EncodedType>, ActualType extends Comparable<ActualType>>
+{
+  /**
+   * Initialize any writer objects that can be instantiated before data from the segments has been processed/merged.
+   * @throws IOException
+   */
+  public void initWriters() throws IOException;
+
+
+  /**
+   * Read the segments to be merged and combine their indexes/data structures internally.
+   * (e.g., merging of dictionaries for String dimensions).
+   *
+   * @param adapters List of adapters to be merged.
+   * @throws IOException
+   */
+  public void mergeAcrossSegments(List<IndexableAdapter> adapters) throws IOException;
+
+
+  /**
+   * Update the DimensionMerger's internal structures with row value(s) to be merged.
+   *
+   * An example implementation might be to add the values to an open column writer.
+   *
+   * @param values The row values to be added.
+   * @throws IOException
+   */
+  public void addColumnValue(Object values) throws IOException;
+
+
+  /**
+   * Close any open writers and write merged data to disk.
+   *
+   * @param rowNumConversions A list of conversion buffers indexed by adapter number
+   *                          The list will have the same order as the list of adapters passed to mergeAcrossSegments()
+   *                          This buffer can be used to convert the row numbers from individual adapters.
+   * @throws IOException
+   */
+  public void closeWriters(List<IntBuffer> rowNumConversions) throws IOException;
```

Comment:
if `mergeAcrossSegments` is merging the data from individual segments, wouldn't it already have the rowNumConversions as well?